composer/datasets/streaming/download.py

```diff
@@ -66,7 +77,11 @@ def download_from_sftp(remote: str, local: str) -> None:
     with SSHClient() as ssh_client:
         # Connect SSH Client
         ssh_client.set_missing_host_key_policy(AutoAddPolicy)
```

Comment:
Ooof ... I missed this line before, but do not think we should default to this. OK making this an option, but would strongly prefer to default to the RejectPolicy (as does paramiko).

CC @nlsapp @hanlint  for security review

Reply:
Ok, is this the same behavior with SFTPObjectStore? Like do we have host keys setup in the docker containers? I can test it.

Reply:
The SFTPObjectStore defaults to reject, but has a config option to override this.

Reply:
Should it reject by default and force the user to provide host keys as an argument to mcli (or like, an env var secret/configmap): Yes

Should we have an option, that allows a user to decide if hostkeys are ignored: Also yes

But def not the default and ideally we should warn them: something something "You are ignoring SFTP Host Keys, THIS IS UNSAFE!" or something

Reply:
Gonna address this in a followup PR, needs some coordination with the SFTPObjectStore and launch interface.